src/java/components/hu.bme.mit.inf.modes3.components.dashboard/src/main/java/hu/bme/mit/inf/modes3/components/dashboard/service/StateChangeService.java

```diff
@@ -57,7 +62,17 @@ public void onTurnoutStateChange(int id, TurnoutState oldValue, TurnoutState new
 	@Override
 	public void onTrainReferenceSpeed(int id, int speed, TrainDirection direction) {
 		Utils.sendTrainReferenceSpeedChange(metaBroadcaster, id, speed, direction);
+	}
 
+	@Override
+	public void onComputerVisionDetection(List<ComputerVisionInformation> information, long timestamp,
+			long frameindex) {
+	
+		for(ComputerVisionInformation cvInfo: information) {
+			if( cvInfo.getName().contains("SNCF") || cvInfo.getName().contains("Taurus") || cvInfo.getName().equals("BR294")) {
```

Comment:
The hard-coded "SNCF, "Taurus", "BR294" constants could be refactored into an array/Set, and an iteration could find out if the `cvInfo.getName().contains(..)` any of them.

Having said that:

```[java]
String[] locomotives = new String[]{"SNCF", "Taurus", "BR294"};
for(ComputerVisionInformation cvInfo: information) {
    for(String locomotive: locomotives){
      if(cvInfo.getName().contains(locomotive)){
        Utils.sendComputerVisionState(metaBroadcaster, cvInfo);
        break;
    }
  }
}
```

Reply:
These names are unique identifiers of each locomotive (as far as I know), thus checking for containment is unnecessary. Check commit 7c3584b for my solution.

Reply:
Yes, but it is easier to add new locomotives. Or one could say, just simple get every locomotive name from the config class: `LocomotivesConfiguration.getLocomotiveNames()` returns every locomotive that is used in the system. However, they will have lowercased names.

Reply:
Oh, sorry . :-) Then simply replace the `Arrays.asList("Taurus", "BR294", "SNCF");` with the names that I suggested. Or if you really rely on that naming, then leave it as it is. (The suggestion would just remove unnecessary redundancy in code.)

Reply:
These names are received from the CV component, and if we do not rename them in that component too, then no locomotive will appear on the dashboard. I'm strongly suggest that this configuration is a working solution for integrating the CV component with the dashboard, leaving it in this state will not break the currently working system.

Reply:
Ok, I totally agree with that.